Directory.Packages.props

```diff
@@ -23,7 +23,17 @@
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="[3.1.0,)" />
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="[3.1.0,)" />
     <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="[2.1.0,)" />
-    <PackageVersion Include="Microsoft.Extensions.Logging" Version="[3.1.0,)" />
+
+    <!--
+        Typically, the latest stable version of Microsoft.Extensions.Logging should be used here because:
+        1) Each major version bump will have some new API capabilities (e.g. .NET 6 introduced compile-time logging
+          source generation, .NET 8 introduced automatic event id generation).
+        2) Each minor version bump is normally security hotfixes or critical bug fixes.
+        3) Since version 3.1.0, the .NET runtime team is holding a high bar for backward compatibility on
+          Microsoft.Extensions.Logging even during major version bumps, so compatibility is not a concern here.
+    -->
+    <PackageVersion Include="Microsoft.Extensions.Logging" Version="8.0.0-rc.1.23419.4" />
```

Comment:
Do we want `Microsoft.Extensions.Logging.Configuration` below to be in sync with `Microsoft.Extensions.Logging` version?

Reply:
Eventually yes. But not in this PR.